scanner/src/main/kotlin/Scanner.kt

```diff
@@ -148,14 +149,24 @@ class Scanner(
         }
 
         val filteredScannerConfig = scannerConfig.copy(options = filteredScannerOptions)
-        val scannerRun = ScannerRun(startTime, endTime, Environment(), filteredScannerConfig, scanResults)
+
+        val scannerRun = ScannerRun(
+            startTime = startTime,
+            endTime = endTime,
+            environment = Environment(),
+            config = filteredScannerConfig,
+            provenances = projectResults.provenances + packageResults.provenances,
+            scanResults = (projectResults.scanResults + packageResults.scanResults).mapTo(mutableSetOf()) {
+                it.clearPackageVerificationCode()
+            }
+        )
 
         return ortResult.copy(scanner = scannerRun)
     }
 
-    suspend fun scan(packages: Set<Package>, context: ScanContext): Map<Identifier, List<ScanResult>> {
+    suspend fun scan(packages: Set<Package>, context: ScanContext): ScannerRun {
```

Comment:
As this is a public function I would have introduced a dedicated class for the return value which only contains the provenance resolution results and the scan results, because only those properties are required. Returning a `ScannerRun` here is misleading when using the function programmatically (e.g. the timestamps are wrong and the config is not filtered unlike with the other scan function).

Reply:
I spent quite a while trying to do that. It was not too easy.

Would you be fine if I addressed this in a follow up PR? (Currently, this function is only called from the test)

Reply:
@mnonnenmacher - I've extracted this to an issue and added it to the corresponding epic, see:
- https://github.com/oss-review-toolkit/ort/issues/7155
- https://github.com/oss-review-toolkit/ort/issues/6945